### DISCUSSION TITLE
Postgresql database privileges

### DISCUSSION THREAD

**Discussion by langeman69** [keycloak/keycloak#25750] [Q&A]
Hi all,

Since most of our databases are already hosted by a PostgreSQL database server, I want to use keycloak using this "external" database. this works fine by creating a datbase and user something like:

> create database keycloak;
> create user keycloak with encrypted password 'my-very-secret-password';
> grant all privileges on database keycloak to keycloak;
> \c keycloak
> grant all on schema public to keycloak;
> 

Of course it will work since the keycloak database user has all privileges. From a security point of view, this might not een good idea and giving the keycloak user _just enough_ privileges to install might be better.
Am I missing something? I cannot find any information on creating a users and the necessary privileges? Anyone got any ideas?

langeman

**Comment by yurios** [comment]
I have the same question.
Hi @langeman69, were you able to figure that out?
Could you please share, it will be much appreciated.
Thank you

**Comment by michael-suissa** [comment]
I'd really appreciate it if anyone who has resolved this would share the information. 
I could of course give keycloak the login/password of postgresql user so that it had all the rights but as langeman said, I'd prefer to give "just enough" privileges.
Thanks.

**Comment by vpsx** [comment]
I am also looking for docs on this

**Comment by J0F3** [comment]
I would also be interested in any recommendations for the necessary privileges on the database.

**Comment by omasseau** [comment]
Indeed it would be useful to document the exact list of necessary privileges. I cannot find anything about it in the current documentation.
